cmd/argocd/commands/cluster.go

```diff
@@ -115,8 +115,9 @@ func NewClusterAddCommand(clientOpts *argocdclient.ClientOptions, pathOpts *clie
 					managerBearerToken, err = clusterauth.GetServiceAccountBearerToken(clientset, clusterOpts.SystemNamespace, clusterOpts.ServiceAccount)
 				} else {
 					isTerminal := isatty.IsTerminal(os.Stdout.Fd()) || isatty.IsCygwinTerminal(os.Stdout.Fd())
+					isNamespaced := len(clusterOpts.Namespaces) > 0
 
-					if isTerminal && !skipConfirmation {
+					if isTerminal && !skipConfirmation && !isNamespaced {
```

Comment:
It will still create a ServiceAccount with full namespace-level privileges though, right? Should we just add a second warning/prompt specific to that case?

Reply:
@crenshaw-dev You are right! I made some changes and dropped the `admin` word from the warning message. It now looks like: `...with full cluster level privileges...` or `...with full namespace level privileges...` based on the absence/presence of the namespace flag. Please let me know if that's ok.
Thanks!

Reply:
That looks awesome, thanks!